### PR TITLE
Add "carriers" field to Booking

### PR DESCRIPTION
**Pull request by jaroslav-kubicek**
I propose the idea to add this field, so we can easily retrieve a unique list of involved carriers and e.g. display a proper logo without the necessity to iterate over all `Trip.leg` and filtering on client.

I also updated the `airlines.json` dataset so it's up to date (Busses and trains were missing).

Controversion - should it be named as "carriers" (airlines+buses+trains) or stick with good old airlines?

### CODE REVIEW

**Comment by mrtnzlml** [comment]
> Controversion - should it be named as "carriers" (airlines+buses+trains) or stick with good old airlines?

I think "carriers" is a great name.

File: src/booking/types/outputs/BookingInterface.js [hunk 1]
```diff
@@ -94,6 +97,32 @@ export const commonFields = {
     resolve: BookingDestinationImageURL,
   },
 
+  carriers: {
+    type: new GraphQLList(Airline),
```
Comment (mrtnzlml) [inline src/booking/types/outputs/BookingInterface.js]: We should probably also focus on this since `Airline` is not accurate as you pointed out.

Comment (mrtnzlml) [reply src/booking/types/outputs/BookingInterface.js]: I am just not sure if it's a good idea to create a new field (`carriers`) with a type that is already virtually deprecated (`Airline`). It's going to be BC break as well to change it later so this field is already doomed and you should think about the future how are you going to replace it. My only advice is: do not kill your nice `carriers` before you even start... :)

Comment (jaroslav-kubicek) [reply src/booking/types/outputs/BookingInterface.js]: updated, it's still similar to `Airline`, as there is not much to add
- id field added
- logoUrl removed (displaying logo works differently now)

**Comment by michalsanger** [comment]
@jaroslav-kubicek fix the conflict plz

**Comment by jaroslav-kubicek** [comment]
@michalsanger  updated

File: src/booking/types/outputs/Carrier.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      description: 'Indicates whether it is low cost carrier.',
+      resolve: ({ isLowCost }: CarrierData): boolean => isLowCost,
+    },
+  },
+});
+
+export const getUniqueCarriers = (
+  carriers: $ReadOnlyArray<?Airline | ?CarrierData>,
+): CarrierData[] =>
+  carriers.reduce((acc, carrier) => {
```
Comment (michalsanger) [inline src/booking/types/outputs/Carrier.js]: When did you fell in love with `reduce` please? :-)

Can't we just use `uniqBy` from Rambda or Lodash? Or use `Set` as registry of know codes and simply iterate the `carriers` array?

File: src/flight/dataloaders/Airline.js [hunk 3]
```diff
@@ -37,7 +37,7 @@ async function fetchAirlines() {
       // eslint-disable-next-line promise/prefer-await-to-then
       .then(airlines => {
         const cache: { [key: string]: Airline } = {};
-        airlines.map((a: Airline) => (cache[a.id] = a));
+        airlines.forEach((a: Airline) => (cache[a.id] = a));
```
Comment (michalsanger) [inline src/flight/dataloaders/Airline.js]: Is there any difference?

Comment (jaroslav-kubicek) [reply src/flight/dataloaders/Airline.js]: sry, I couldn't resist :) 

map => getting new array from the old one
forEach => doing some side effects for each item in the array

File: src/booking/types/outputs/Carrier.js (hunk 2 above)
Comment (jaroslav-kubicek) [reply src/booking/types/outputs/Carrier.js]: When I heard about the functional approach in programming for the first time.

reduce is normal, quite convenient function, no need to have dozens of immediate variables to do the same thing.

File: src/flight/dataloaders/Airline.js (hunk 3 above)
Comment (jaroslav-kubicek) [reply src/flight/dataloaders/Airline.js]: it could be refactored to reduce as well btw... 👼

**Pull request merged**